src/ia-account-settings.interface.ts

```diff
@@ -0,0 +1,81 @@
+export interface IAAccountSettingsInterface {
+  /**
+   * set screenname in UserModel
+   *
+   * @param {Event} e
+   * @memberof IAAccountSettingsInterface
+   */
+  setScreenname(e: Event): void;
+
+  /**
+   * set email in UserModel
+   *
+   * @param {Event} e
+   * @memberof IAAccountSettingsInterface
+   */
+  setEmail(e: Event): void;
+
+  /**
+   * set password in UserModel
+   *
+   * @param {Event} e
+   * @memberof IAAccountSettingsInterface
+   */
+  setPassword(e: Event): void;
+
+  /**
+   * set user loan history visiblility preferences
+   *
+   * @param {Event} e
+   * @memberof IAAccountSettingsInterface
+   */
+  setBorrowHistory(e: Event): void;
+
+  /**
+   * check if fields has errors
+   *
+   * @return {*}  {boolean}
+   * @memberof IAAccountSettingsInterface
+   */
+  hasFieldError(): boolean;
+
+  /**
+   * reset error fields
+   * @param {string} value
+   * @memberof IAAccountSettingsInterface
+   */
+  resetErrorFields(value: string): void;
+
+  /**
+   * validate the screenname and set error in ErrorModel
+   *
+   * @return {*}  {Promise<void>}
+   * @memberof IAAccountSettingsInterface
+   */
+  validateScreenname(): Promise<void>;
+
+  /**
+   * return true if screenname is available
+   *
+   * @return {*}  {boolean}
+   * @memberof IAAccountSettingsInterface
+   */
+  isScreennameAvailable(): Promise<boolean | undefined>;
+
+  /**
+   * validate the email and set error in ErrorModel
+   *
+   * @return {*}  {Promise<void>}
+   * @memberof IAAccountSettingsInterface
+   */
+  validateEmail(): Promise<void>;
+
+  /**
+   * after validation, save the account setting
+   *
+   * @param {Event} e
+   * @return {*}  {Promise<void>}
```

Comment:
suggestion: remove jsdoc since we can inline the param and return types 🙂

Reply:
I wanted to consolidated at single place for all the functions I am using so added all the jsdocs here only.